python/feathub/feature_views/transforms/tests/test_sliding_window_transform.py

```diff
@@ -1261,6 +1261,103 @@ def test_join_sliding_feature(self):
                 f"actual: {result_df}",
             )
 
+    def test_sliding_feature_join_source(self):
+        df = pd.DataFrame(
+            [
+                ["Alex", 100.0, "2022-01-01 09:01:00"],
+                ["Alex", 200.0, "2022-01-01 09:01:20"],
+                ["Alex", 450.0, "2022-01-01 09:06:00"],
+            ],
+            columns=["name", "cost", "time"],
+        )
+
+        schema = Schema(["name", "cost", "time"], [String, Float64, String])
+        source = self.create_file_source(
+            df, schema=schema, keys=["name"], name="source"
+        )
+
+        df2 = pd.DataFrame(
+            [
+                ["Alex", "a", "2022-01-01 09:01:00"],
+                ["Alex", "b", "2022-01-01 09:02:00"],
+                ["Alex", "c", "2022-01-01 09:05:00"],
+                ["Alex", "d", "2022-01-01 09:07:00"],
+                ["Alex", "e", "2022-01-01 09:09:00"],
+            ]
+        )
+        source2 = self.create_file_source(
+            df2,
+            schema=Schema(["name", "feature", "time"], [String, String, String]),
+            keys=["name"],
+            name="source2",
+            timestamp_field="time",
+        )
+
+        features = SlidingFeatureView(
+            name="features",
+            source=source,
+            features=[
+                Feature(
+                    name="last_2_minute_total_cost",
+                    transform=SlidingWindowTransform(
+                        expr="cost",
+                        agg_func="SUM",
+                        group_by_keys=["name"],
+                        window_size=timedelta(minutes=2),
+                        step_size=timedelta(minutes=1),
+                    ),
+                ),
+                Feature(
+                    name="cnt",
+                    transform=SlidingWindowTransform(
+                        expr="1",
+                        agg_func="COUNT",
+                        group_by_keys=["name"],
+                        window_size=timedelta(minutes=2),
+                        step_size=timedelta(minutes=1),
+                    ),
+                ),
+            ],
+        )
+
+        joined_feature = DerivedFeatureView(
```

Comment:
Can we make the variable names more intuitive by using `*features*` to name instances of FeatureView?